### PR TITLE
Upgraded OnnxRuntime to v1.0 and Google Protobuf to 3.10.1

### PR DESCRIPTION
**Pull request by harishsk**
This PR upgrades the OnnxRuntime to v1.0 and the Google Protobuf version to 3.10.1

### CODE REVIEW

File: build/Dependencies.props [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
   <!-- Other/Non-Core Product Dependencies -->
   <PropertyGroup>
-    <GoogleProtobufPackageVersion>3.5.1</GoogleProtobufPackageVersion>
+    <GoogleProtobufPackageVersion>3.10.1</GoogleProtobufPackageVersion>
     <LightGBMPackageVersion>2.2.3</LightGBMPackageVersion>
     <MicrosoftExtensionsPackageVersion>2.1.0</MicrosoftExtensionsPackageVersion>
-    <MicrosoftMLOnnxRuntimePackageVersion>0.5.1</MicrosoftMLOnnxRuntimePackageVersion>
+    <MicrosoftMLOnnxRuntimePackageVersion>1.0</MicrosoftMLOnnxRuntimePackageVersion>
```
Comment (eerhardt) [inline build/Dependencies.props]: This should be `1.0.0`

File: src/Microsoft.ML.OnnxTransformer/Microsoft.ML.OnnxTransformer.csproj [hunk 2]
```diff
@@ -9,7 +9,7 @@
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.ML.Core\Microsoft.ML.Core.csproj" />
     <ProjectReference Include="..\Microsoft.ML.Data\Microsoft.ML.Data.csproj" />
-    <PackageReference Include="Microsoft.ML.OnnxRuntime" Version="$(MicrosoftMLOnnxRuntimePackageVersion)" />
+    <PackageReference Include="Microsoft.ML.OnnxRuntime.Gpu" Version="$(MicrosoftMLOnnxRuntimePackageVersion)" />
```
Comment (eerhardt) [inline src/Microsoft.ML.OnnxTransformer/Microsoft.ML.OnnxTransformer.csproj]: We can't change this. Or else people without GPUs won't be able to use the OnnxTransform. See [https://github.com/microsoft/onnxruntime/issues/2184](https://github.com/microsoft/onnxruntime/issues/2184)

File: build/Dependencies.props (hunk 1 above)
Comment (harishsk) [reply build/Dependencies.props]: Good catch. Fixed. Thanks!

---
In reply to: [340776622](https://github.com/dotnet/machinelearning/pull/4416#discussion_r340776622) [](ancestors = 340776622)

File: src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs [hunk 3]
```diff
@@ -439,7 +441,15 @@ public static NamedOnnxValue CreateScalarNamedOnnxValue<T>(string name, T data)
         {
             if (!_onnxTypeMap.Contains(typeof(T)))
                 throw new NotImplementedException($"Not implemented type {typeof(T)}");
-            return NamedOnnxValue.CreateFromTensor<T>(name, new DenseTensor<T>(new T[] { data }, new int[] { 1 }));
+
+            if (data.GetType() == typeof(ReadOnlyMemory<Char>))
```
Comment (eerhardt) [inline src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs]: Should be `if (typeof(T) == typeof(ReadOnlyMemory<Char>))`. This allows the compiler/runtime to remove this check when `typeof(T)` is `int`, `float`, etc.

File: src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs [hunk 4]
```diff
@@ -439,7 +441,15 @@ public static NamedOnnxValue CreateScalarNamedOnnxValue<T>(string name, T data)
...
+            {
+                var charMemory = (ReadOnlyMemory<Char>)Convert.ChangeType(data, typeof(ReadOnlyMemory<Char>));
```
Comment (eerhardt) [inline src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs]: You don't need `Convert.ChangeType`. You can just cast `data` to `ReadOnlyMemory<char>`.

File: src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (harishsk) [reply src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs]: Sounds good. Will fix in next version.

---
In reply to: [340779853](https://github.com/dotnet/machinelearning/pull/4416#discussion_r340779853) [](ancestors = 340779853)

File: src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs (hunk 4 above)
Comment (harishsk) [reply src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs]: That doesn't seem to work for me. I get this error:

Cannot convert type 'T' to 'System.ReadOnlyMemory<char>'

---
In reply to: [340779381](https://github.com/dotnet/machinelearning/pull/4416#discussion_r340779381) [](ancestors = 340779381)

File: src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs [hunk 5]
```diff
@@ -439,7 +441,15 @@ public static NamedOnnxValue CreateScalarNamedOnnxValue<T>(string name, T data)
...
+                var stringMemory = new Memory<string>(new string[] { charMemory.ToString() });
```
[the remark this answers is not in the document]
Comment (harishsk) [reply src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs]: The updated version works. I have pushed the updated fix. Thanks for reviewing this specifically.

---
In reply to: [340789987](https://github.com/dotnet/machinelearning/pull/4416#discussion_r340789987) [](ancestors = 340789987)

**Comment by ganik** [comment]
LGTM. thx

**Pull request merged**